### PR TITLE
Tpbot/Add initial structure for server unit

### PR DESCRIPTION
**Pull request by Mecharyry**
#### Problem
We need to have a middleman server to send messages between a bot and a client. Given that I haven't done a lot of JavaScript / node / socketIO in the past, I want to keep the changes small for quick, accurate feedback.

#### Solution
This PR creates the initial structure for the server-unit of the tpbot app.

### CODE REVIEW

File: TelepresenceBot/server-unit/core/server.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+var clients = {};
+
+io.sockets.on('connection', function (client) {
+
+    console.log('connecting: ' + client.id);
+    clients[client.id] = client;
+    io.sockets.emit('connected', toKeysArrayFrom(clients));
+
+
```
Comment (danybony) [inline TelepresenceBot/server-unit/core/server.js]: One line is enough

File: TelepresenceBot/server-unit/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "main": "server.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test": "mocha"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/novoda/spikes/tree/tpbot-feature-branch"
```
Comment (danybony) [inline TelepresenceBot/server-unit/package.json]: Maybe we could just put the repo URL already, as we hopefully will merge soon to master

File: TelepresenceBot/server-unit/test/serverTest.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+var socketURL = 'http://0.0.0.0:5000'
+
+var options ={
+    transports: ['websocket'],
+    'force new connection': true
+};
+
+describe("TelepresenceBot Server: DummyTest ",function() {
+
+    it('Should add client to list of clients on connection.', function(done) {
```
Comment (danybony) [inline TelepresenceBot/server-unit/test/serverTest.js]: Can you test the disconnection as well?

File: TelepresenceBot/server-unit/core/server.js (hunk 1 above)
Comment (Mecharyry) [inline TelepresenceBot/server-unit/core/server.js]: Oh my, how did that sneak in, kill it with 🔥  1a403a7

File: TelepresenceBot/server-unit/package.json (hunk 2 above)
Comment (Mecharyry) [inline TelepresenceBot/server-unit/package.json]: WE WILL BE IN MASTER!!! e611167

File: TelepresenceBot/server-unit/test/serverTest.js (hunk 3 above)
Comment (Mecharyry) [inline TelepresenceBot/server-unit/test/serverTest.js]: Ah yes, I just wanted to set up and initial structure but I should have tested this dummy code better 😆 5e994ec

**Pull request merged**